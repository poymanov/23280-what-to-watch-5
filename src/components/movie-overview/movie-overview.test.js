import React from "react";
import renderer from "react-test-renderer";
import MovieOverview from "./movie-overview";
import {extend} from "../../utils/utils";

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

const movieWithBadRating = extend(movie, {rating: 3});
const movieWithNormalRating = extend(movie, {rating: 5});
const movieWithGoodRating = extend(movie, {rating: 8});
const movieWithVeryGoodRating = extend(movie, {rating: 9});
const movieWithAwesomeRating = extend(movie, {rating: 10});

describe(`Should MovieOverview render correctly`, () => {
  it(`With bad rating`, () => {
    const tree = renderer
      .create(<MovieOverview movie={movieWithBadRating}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With normal rating`, () => {
    const tree = renderer
      .create(<MovieOverview movie={movieWithNormalRating}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With good rating`, () => {
    const tree = renderer
      .create(<MovieOverview movie={movieWithGoodRating}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With very good rating`, () => {
    const tree = renderer
      .create(<MovieOverview movie={movieWithVeryGoodRating}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With very awesome rating`, () => {
    const tree = renderer
      .create(<MovieOverview movie={movieWithAwesomeRating}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


