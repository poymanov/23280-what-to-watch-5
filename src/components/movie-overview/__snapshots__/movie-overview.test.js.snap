// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieOverview render correctly With bad rating 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      3
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;

exports[`Should MovieOverview render correctly With good rating 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;

exports[`Should MovieOverview render correctly With normal rating 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;

exports[`Should MovieOverview render correctly With very awesome rating 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      10
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Awesome
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;

exports[`Should MovieOverview render correctly With very good rating 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      test
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        test
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        test
      </strong>
    </p>
  </div>,
]
`;
